osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs

```diff
@@ -305,18 +305,34 @@ public override bool OnBackButton()
                 return true;
             }
 
+            return base.OnBackButton();
+        }
+
+        public override bool OnExiting(IScreen next)
+        {
+            if (client.Room == null)
+            {
+                // room has not been created yet; exit immediately.
+                return base.OnExiting(next);
+            }
+
             if (!exitConfirmed && dialogOverlay != null)
             {
-                dialogOverlay.Push(new ConfirmDialog("Are you sure you want to leave this multiplayer match?", () =>
+                if (dialogOverlay.CurrentDialog is ConfirmDialog confirmDialog)
+                    confirmDialog.PerformOkAction();
```

Comment:
Was there a specific reason for changing this to pushing the dialog button instead of proceeding with exit? Just checking because in theory they should both behave the same way.

Reply:
No reason. I just copied what main menu and editor did.

Reply:
Ah, fair enough. I think it's fine to leave like this as it makes it very explicit what is going on.